Animations/CMakeLists.txt

```diff
@@ -0,0 +1,220 @@
+        src/editor/EditorCameraController.cpp
+        src/editor/MenuBar.cpp
+        src/editor/EditorGui.cpp
+        src/editor/ImGuiTimeline.cpp
+        src/editor/DebugPanel.cpp
+        src/editor/SceneHierarchyPanel.cpp
+        src/editor/ExportPanel.cpp
+        src/editor/AnimObjectPanel.cpp
+        src/editor/EditorSettings.cpp
+        src/editor/ConsoleLog.cpp
+        src/editor/SceneManagementPanel.cpp
+        src/editor/AssetManagerPanel.cpp
+        src/editor/Gizmos.cpp
+
+        src/svg/Paths.cpp
+        src/svg/SvgCache.cpp
+        src/svg/Svg.cpp
+        src/svg/SvgParser.cpp
+
+        src/renderer/Shader.cpp
+        src/renderer/PerspectiveCamera.cpp
+        src/renderer/Renderer.cpp
+        src/renderer/Texture.cpp
+        src/renderer/Colors.cpp
+        src/renderer/Fonts.cpp
+        src/renderer/OrthoCamera.cpp
+        src/renderer/GLApi.cpp
+        src/renderer/Framebuffer.cpp
+
+        src/utils/CMath.cpp
+        src/utils/TableOfContents.cpp
+
+        src/math/DataStructures.cpp
+
+        src/platform/FileSystemWatcherWin32.cpp
+        src/platform/PlatformLinux.cpp
+        src/platform/FileSystemWatcherLinux.cpp
+        src/platform/PlatformWin32.cpp
+
+        src/core/ImGuiLayer.cpp
+        src/core/Input.cpp
+        src/core/ProjectApp.cpp
+        src/core/Application.cpp
+        src/core/GladLayer.cpp
+        src/core/Window.cpp
+
+        src/latex/LaTexLayer.cpp
+
+        src/SingleIncludeImpls.cpp
+        src/core.cpp
+        src/main.cpp)
+
+if(MATH_ANIMATION_OS_LINUX)
+    set(MATH_ANIMATIONS_PLATFORM_SOURCE
+            src/platform/PlatformLinux.cpp
+            src/platform/FileSystemWatcherLinux.cpp)
+
+    # TODO: The source code used to use _MAX_PATH, but this can cause issues on Linux
+    # see: https://stackoverflow.com/questions/833291/is-there-an-equivalent-to-winapis-max-path-under-linux-unix
+    set(MATH_ANIMATIONS_MAX_PATH 4096)
+elseif(MATH_ANIMATION_OS_WINDOWS)
+    set(MATH_ANIMATIONS_PLATFORM_SOURCE
+            src/platform/FileSystemWatcherWin32.cpp
+            src/platform/PlatformWin32.cpp)
+
+    # Alias to the macro _MAX_PATH on windows
+    set(MATH_ANIMATIONS_MAX_PATH _MAX_PATH)
+else()
+    message(FATAL_ERROR "Targeting unsupported operating system ${CMAKE_SYSTEM_NAME}")
+endif()
+
+add_executable(MathAnimations ${MATH_ANIMATIONS_SOURCE} ${MATH_ANIMATIONS_PLATFORM_SOURCE})
+target_include_directories(MathAnimations PUBLIC ${MATH_ANIMATIONS_INCLUDE_DIR})
+target_compile_definitions(MathAnimations PUBLIC
+        MATH_ANIMATIONS_MAX_PATH=${MATH_ANIMATIONS_MAX_PATH}
+)
+target_compile_definitions(MathAnimations PRIVATE
+        $<$<CONFIG:Debug>:_DEBUG=1>
+        $<$<CONFIG:RelWithDebInfo>:_RELEASE=1>
+        $<$<CONFIG:MinSizeRel>:_RELEASE=1>
+        $<$<CONFIG:Release>:_RELEASE=1>
+)
+source_group(TREE "${CMAKE_CURRENT_SOURCE_DIR}" FILES ${MATH_ANIMATIONS_SOURCE} ${MATH_ANIMATIONS_PLATFORM_SOURCE})
+
+target_link_libraries(MathAnimations PUBLIC
```

Comment:
It looks like you're missing this option:

```cmake
if (MSVC) 
    set_target_properties(
        MathAnimations
        NativeFileDialog
        Glad
        DearImGui
        DearImGuiBackend
        tinyxml2
        plutovg
        OpenAL
        freetype
        common 
        ex-common

        PROPERTIES
        MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>"
    )
endif()
```
This is important on MSVC to ensure a consistent c-runtime library is used for the static linking across all the libraries.

Reply:
This is now set as a CMake option for the entire project including all subprojects, see top-level CMakeLists.txt line  48:
```cmake
if(MSVC)
    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
endif()
```